src/main/groovy/pl/allegro/tech/build/axion/release/ReleaseTask.groovy

```diff
@@ -15,10 +16,21 @@ abstract class ReleaseTask extends BaseAxionTask {
     void release() {
         VersionResolutionContext context = resolutionContext()
         Releaser releaser = context.releaser()
+        ScmService scmService = context.scmService()
+
+        if (scmService.isReleaseOnlyOnDefaultBranches()) {
+            def releaseBranchNames = scmService.getReleaseBranchNames()
+            def currentBranch = context.repository().currentPosition().getBranch()
+            if (!releaseBranchNames.contains(currentBranch)) {
+                logger.quiet("Release step skipped since 'releaseOnlyOnDefaultBranches' option is set, and '${currentBranch}' was not in 'releaseBranchNames' list [${releaseBranchNames.join(', ')}]")
+                return
+            }
+        }
```

Comment:
Shouldn't the `createRelease` and `pushRelease` tasks have the same conditional logic to be noop on non-release branches? Maybe it should be moved to `Releaser` class? That would also allow us to unit test this behavior in `ReleaserTest` where it would be easier to verify whether the tag was really created or not.

Reply:
yes, this is a good point, I've moved code to Releser class :)